synapseclient/api/json_schema_services.py

```diff
@@ -207,15 +265,399 @@ async def get_json_schema_derived_keys(
+
+    Updates the permissions for the specified organization. The caller must have
+    ACCESS_TYPE.CHANGE_PERMISSIONS permission on the organization. The etag from
+    a previous get_organization_acl() call is required for concurrency control.
+
+    Arguments:
+        organization_id: The numeric identifier of the organization
+        resource_access: List of ResourceAccess objects, each containing:
+                        - principalId: The user or team ID
+                        - accessType: List of permission types (e.g., ["READ", "CREATE", "DELETE"])
+        etag: The etag from get_organization_acl() for concurrency control
+        synapse_client: If not passed in and caching was not disabled by
+                       `Synapse.allow_client_caching(False)` this will use the last created
+                       instance from the Synapse class constructor
+
+    Returns:
+        The updated AccessControlList object matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/AccessControlList.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    request_body = {"resourceAccess": resource_access, "etag": etag}
+
+    return await client.rest_put_async(
+        uri=f"/schema/organization/{organization_id}/acl", body=json.dumps(request_body)
+    )
+
+
+async def list_json_schemas(
+    organization_name: str, *, synapse_client: Optional["Synapse"] = None
+) -> List[Dict[str, Any]]:
+    """
+    List all JSON schemas for an organization.
+
+    Retrieves all JSON schemas that belong to the specified organization. This operation
+    does not require authentication and will return all publicly visible schemas.
+
+    Arguments:
+        organization_name: The name of the organization to list schemas for
+        synapse_client: If not passed in and caching was not disabled by
+                       `Synapse.allow_client_caching(False)` this will use the last created
+                       instance from the Synapse class constructor
+
+    Returns:
+        A list of JsonSchemaInfo objects, each containing:
+        - organizationId: The Synapse issued numeric identifier for the organization.
+        - organizationName: The name of the organization to which this schema belongs.
+        - schemaId: The Synapse issued numeric identifier for the schema.
+        - schemaName: The name of the this schema.
+        - createdOn: The date this JSON schema was created.
+        - createdBy: The ID of the user that created this JSON schema.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    request_body = {"organizationName": organization_name}
+
+    results = []
+
+    async for item in rest_post_paginated_async(
+        "/schema/list", body=request_body, synapse_client=client
+    ):
+        results.append(item)
+
+    return results
+
+
+async def list_json_schema_versions(
```

Comment:
@BryanFauble Same for this function. Based on the documentation [here](https://rest-docs.synapse.org/rest/POST/schema/version/list.html), this page should return a single page of result, not everything

Reply:
Corrected in https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1228